React/Modules/RCTEventEmitter.m

```diff
@@ -65,7 +65,7 @@ - (void)sendEventWithName:(NSString *)eventName body:(id)body
         [[self supportedEvents] componentsJoinedByString:@"`, `"]);
   }
 
-  BOOL shouldEmitEvent = (_observationDisabled || _listenerCount > 0);
+  BOOL shouldEmitEvent = (!_observationDisabled && _listenerCount > 0);
```

Comment:
When native side observation is disabled, _listenerCount will always be 0. But that doesn't mean that the JS side of this event emitter doesn't have any observers. That's why we should emit the event when observation is disabled, or when the listener count is > 0.